instructorFeedbackResults: Give a 'publish' button in the results page. #2659

tangzijian opened the pull request:
Fixes #2659

LowWeiLin commented:
@tangzijian Could you post a screenshot of the changes? Thanks!

tangzijian commented:
@LowWeiLin 
![screen shot 2015-03-21 at 12 20 35 pm](https://cloud.githubusercontent.com/assets/4153394/6763727/caf51c08-cfc4-11e4-9f67-d31d4982947c.png)

LowWeiLin commented:
@damithc does this look okay?

damithc commented:
Looks ok. May be use 'Publish results' to be consistent with the other button?

tangzijian commented:
@damithc 
Dr.Damith,
Should I make the button have the same width with 'Download results' button?
![screen shot 2015-03-21 at 2 34 27 pm](https://cloud.githubusercontent.com/assets/4153394/6764050/b29e20e2-cfd7-11e4-8efc-d7c4fc466d2f.png)
![screen shot 2015-03-21 at 2 35 12 pm](https://cloud.githubusercontent.com/assets/4153394/6764051/b8cbc532-cfd7-11e4-8422-e1c65d5c8027.png)

damithc commented:
Yes, same width is better, if you can manage it easily.

tangzijian commented:
Same button width.
![screen shot 2015-03-21 at 3 03 04 pm](https://cloud.githubusercontent.com/assets/4153394/6764120/807ac274-cfdb-11e4-916c-94ca5c0094a1.png)

damithc commented:
Looks good

tangzijian commented:
I have done the modification and pushed the commits.

kanghj commented on src/main/webapp/jsp/instructorFeedbackResultsTop.jsp:
```diff
@@ -51,10 +51,10 @@
           <div class="col-sm-2">
               
-              <div class="form-group">
+              <div class="form-group"  style="padding-right:15px;">
```
Maybe try to use bootstrap to align things instead? padding-right may cause alignment issues when the page is viewed on smaller screens.

tangzijian commented:
Sorry for the late reply. I have replaced the css padding with bootstrap.

kanghj commented on src/test/resources/pages/godmode.html:
```diff
@@ -15,8 +15,8 @@
             id="start"
-            value="<!-- DATETODAY -->">
-        <p class="form-control-static"><!-- DATETIMETODAY --></p>
+            value="23/03/2015">
```
Remove the files unrelated to the issue

kanghj commented on src/test/resources/pages/newlyJoinedInstructorFeedbackResultsPage.html:
```diff
@@ -52,17 +52,23 @@
           <div class="col-sm-2">
               
-              <div class="form-group">
+              <div class="form-group" style="padding-right:15px;">
```
There's still a padding-right here.

tangzijian commented:
I modified expected output already, is it ok now?

kanghj commented on src/main/java/teammates/ui/controller/PageData.java:
```diff
@@ -823,7 +823,7 @@ public static String getInstructorHoverMessageForFeedbackSession(FeedbackSession
     
     /**
-     * Returns the links of actions available for a specific evaluation
+     * Returns the link of publish action for a session feedback
```
Is this comment correct?

kanghj commented:
You still have to remove the unrelated changes (godmode.html).

tangzijian commented:
Hi, Mr.Kang Hong Jin,
I have removed the unrelated changes and corrected the java doc for the method. Sorry for the mis-committed file (godmode.html). I used 'git commit -a', it should have not commit the untracked file godmode.html. I will be more careful next time.

kanghj commented on src/test/resources/pages/newlyJoinedInstructorFeedbackResultsPage.html:
```diff
@@ -53,16 +53,24 @@
+                      </form><br>
+                      
+                      <div>
+                      <a class="btn btn-primary btn-block btn-tm-actions session-unpublish-for-test" data-original-title="Make responses no longer visible" data-placement="top" data-toggle="tooltip" href="/page/instructorFeedbackUnpublish?courseid=AHPUiT.instr1.gma-demo&amp;fsname=Second+team+feedback+session&amp;next=%2Fpage%2FinstructorFeedbacksPage%3Fuser%3Dinstructor&amp;user=instructor" oncl ...
```
I am not able to pass all tests, the "instructor" user in this href should be {$test.instructor} instead.

tangzijian replied:
How come I passed all the tests? I go check first

kanghj replied:
{$test.instructor} will be replaced during testing by the value indicated in test.properties.

kanghj commented on src/main/java/teammates/ui/controller/PageData.java:
```diff
@@ -912,27 +908,51 @@ public String getInstructorFeedbackSessionActions(FeedbackSessionAttributes sess
+     *         A string containing the CSS styles of the unpublish link
+     * @return
+     */
+    public String getInstructorFeedbackSessionPublishAndUnpublishAction(FeedbackSessionAttributes session, boolean isHome, InstructorAttributes instructor, String publishLinkStyle, String unpublishLinkStyle) {
```
Sorry for not mentioning this earlier, but I am not so sure if passing the css styles as a parameter into the function is a good idea. Maybe just define these variables in the function itself?

tangzijian replied:
But they have different class attributes in different places. I think trying to define those variables in the function itself will not work. At least we need to pass in one argument to say which type of button we want to create so the correct class attributes can be appended. Or we can define two methods to handle this, but it seems not very good neither. Can you give me some suggestion?

kanghj replied:
Since the button is displayed differently on different pages, maybe implement another function/override the function in the PageData class for the Results Page?

tangzijian replied:
Yeah. I think override is the best way. Thanks for the tip

kanghj commented:
Some changes requested. Good work on changing the html test files across the different tests.

tangzijian commented:
Hi Hong Jin,
I have modified the method as you mentioned earlier.

kanghj commented:
@tangzijian, the test is still not passing for me. There is another instance of "instructor" to change to ${test.instructor}. One way to check if the changes to the test is right is to change the values of the names in test.properties and run the test again.

tangzijian commented:
@kanghj I have done the requested changes.

kanghj commented:
Ok. Looks good now. :)
